File: flutter_app/lib/screens/steps/education_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';

import '../../models/cv_data.dart';
import '../../state/cv_model.dart';
import '../../theme/colors.dart';
import '../../widgets/cv_text_field.dart';

/// Port de `ui/screens/EducationScreen.kt`.
class EducationScreen extends StatefulWidget {
  const EducationScreen({super.key});

  @override
  State<EducationScreen> createState() => _EducationScreenState();
}

class _EducationScreenState extends State<EducationScreen> {
  late bool _showForm;

  @override
  void initState() {
    super.initState();
    _showForm = context.read<CvModel>().cvData.educations.isEmpty;
  }

  @override
  Widget build(BuildContext context) {
    final model = context.watch<CvModel>();
    final scheme = Theme.of(context).colorScheme;
    final educations = model.cvData.educations;
    final showForm = _showForm;

    return SingleChildScrollView(
      padding: const EdgeInsets.symmetric(horizontal: 20, vertical: 16),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.stretch,
        children: [
          const Text(
            'Formation',
            style: TextStyle(
              fontSize: 20,
              fontWeight: FontWeight.bold,
              color: AppColors.gold,
            ),
          ),
          const SizedBox(height: 16),
          for (final edu in educations) ...[
            Card(
              color: AppColors.mediumGray,
              shape: RoundedRectangleBorder(
                borderRadius: BorderRadius.circular(12),
              ),
              child: Padding(
                padding: const EdgeInsets.all(16),
                child: Row(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Expanded(
                      child: Column(
                        crossAxisAlignment: CrossAxisAlignment.start,
                        children: [
                          Text(
                            edu.degree,
                            style: const TextStyle(
                              fontWeight: FontWeight.w600,
                              color: AppColors.gold,
                            ),
                          ),
                          Text(edu.school,
                              style: const TextStyle(fontSize: 14)),
                          if (edu.field.isNotEmpty)
                            Text(
                              edu.field,
                              style: TextStyle(
                                fontSize: 13,
                                color: scheme.onSurfaceVariant,
                              ),
                            ),
                          Text(
                            '${edu.startYear} — ${edu.endYear}',
                            style: TextStyle(
                              fontSize: 12,
                              color: scheme.onSurfaceVariant,
                            ),
                          ),
                        ],
                      ),
                    ),
                    IconButton(
                      onPressed: () => model.removeEducation(edu.id),
                      icon: Icon(Icons.delete, color: scheme.error),
                    ),
                  ],
                ),
              ),
            ),
            const SizedBox(height: 16),
          ],
          if (showForm)
            EducationForm(
              onSave: (edu) {
                model.addEducation(edu);
                setState(() => _showForm = false);
              },
              onCancel: () => setState(() => _showForm = false),
            )
          else
            OutlinedButton.icon(
              onPressed: () => setState(() => _showForm = true),
              icon: const Icon(Icons.add, color: AppColors.gold),
              label: const Text(
                'Ajouter une formation',
                style: TextStyle(color: AppColors.gold),
              ),
            ),
          const SizedBox(height: 80),
        ],
      ),
    );
  }
}

class EducationForm extends StatefulWidget {
  const EducationForm({
    super.key,
    required this.onSave,
    required this.onCancel,
  });

  final ValueChanged<Education> onSave;
  final VoidCallback onCancel;

  @override
  State<EducationForm> createState() => _EducationFormState();
}

class _EducationFormState extends State<EducationForm> {
  String _school = '';
  String _degree = '';
  String _field = '';
  String _startYear = '';
  String _endYear = '';

  @override
  Widget build(BuildContext context) {
    final scheme = Theme.of(context).colorScheme;

    return Container(
      decoration: BoxDecoration(
        color: AppColors.mediumGray,
        borderRadius: BorderRadius.circular(12),
      ),
      padding: const EdgeInsets.all(16),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.stretch,
        children: [
          const Text(
            'Nouvelle formation',
            style: TextStyle(
              fontWeight: FontWeight.w600,
              color: AppColors.gold,
            ),
          ),
          const SizedBox(height: 12),
          CvTextField(
            value: _degree,
            label: 'Diplôme *',
            onChanged: (v) => _degree = v,
          ),
          const SizedBox(height: 12),
          CvTextField(
            value: _school,
            label: 'École / Université *',
            onChanged: (v) => _school = v,
          ),
          const SizedBox(height: 12),
          CvTextField(
            value: _field,
            label: 'Domaine / Spécialité',
            onChanged: (v) => _field = v,
          ),
          const SizedBox(height: 12),
          Row(
            children: [
              Expanded(
                child: CvTextField(
                  value: _startYear,
                  label: 'Année début',
                  onChanged: (v) => _startYear = v,
                ),
              ),
              const SizedBox(width: 12),
              Expanded(
                child: CvTextField(
                  value: _endYear,
                  label: 'Année fin',
                  onChanged: (v) => _endYear = v,
                ),
              ),
            ],
          ),
          const SizedBox(height: 12),
          Row(
            children: [
              Expanded(
                child: OutlinedButton(
                  onPressed: widget.onCancel,
                  child: const Text('Annuler'),
                ),
              ),
              const SizedBox(width: 12),
              Expanded(
                child: FilledButton(
                  style: FilledButton.styleFrom(
                    backgroundColor: AppColors.gold,
                    foregroundColor: scheme.onPrimary,
                  ),
                  onPressed: () {
                    if (_degree.isNotEmpty && _school.isNotEmpty) {
                      widget.onSave(
                        Education(
                          school: _school,
                          degree: _degree,
                          field: _field,
                          startYear: _startYear,
                          endYear: _endYear,
                        ),
                      );
                    }
                  },
                  child: const Text('Enregistrer'),
                ),
              ),
            ],
          ),
        ],
      ),
    );
  }
}

File: app/src/main/java/com/cvgenerator/ui/screens/EducationScreen.kt
package com.cvgenerator.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Add
import androidx.compose.material.icons.filled.Delete
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.cvgenerator.data.CvViewModel
import com.cvgenerator.data.Education
import com.cvgenerator.ui.theme.Gold
import com.cvgenerator.ui.theme.MediumGray

@Composable
fun EducationScreen(viewModel: CvViewModel) {
    val cvData by viewModel.cvData.collectAsState()
    var showForm by remember { mutableStateOf(cvData.educations.isEmpty()) }

    Column(
        modifier = Modifier
            .fillMaxSize()
            .verticalScroll(rememberScrollState())
            .padding(horizontal = 20.dp, vertical = 16.dp),
        verticalArrangement = Arrangement.spacedBy(16.dp)
    ) {
        Text("Formation", fontSize = 20.sp, fontWeight = FontWeight.Bold, color = Gold)

        cvData.educations.forEach { edu ->
            Card(
                modifier = Modifier.fillMaxWidth(),
                shape = RoundedCornerShape(12.dp),
                colors = CardDefaults.cardColors(containerColor = MediumGray)
            ) {
                Row(
                    modifier = Modifier.padding(16.dp),
                    horizontalArrangement = Arrangement.SpaceBetween,
                    verticalAlignment = Alignment.Top
                ) {
                    Column(modifier = Modifier.weight(1f)) {
                        Text(edu.degree, fontWeight = FontWeight.SemiBold, color = Gold)
                        Text(edu.school, fontSize = 14.sp)
                        if (edu.field.isNotBlank()) Text(edu.field, fontSize = 13.sp, color = MaterialTheme.colorScheme.onSurfaceVariant)
                        Text("${edu.startYear} — ${edu.endYear}", fontSize = 12.sp, color = MaterialTheme.colorScheme.onSurfaceVariant)
                    }
                    IconButton(onClick = { viewModel.removeEducation(edu.id) }) {
                        Icon(Icons.Default.Delete, contentDescription = null, tint = MaterialTheme.colorScheme.error)
                    }
                }
            }
        }

        if (showForm) {
            EducationForm(
                onSave = { viewModel.addEducation(it); showForm = false },
                onCancel = { showForm = false }
            )
        } else {
            OutlinedButton(
                onClick = { showForm = true },
                modifier = Modifier.fillMaxWidth(),
                colors = ButtonDefaults.outlinedButtonColors(contentColor = Gold)
            ) {
                Icon(Icons.Default.Add, contentDescription = null, tint = Gold)
                Spacer(Modifier.width(8.dp))
                Text("Ajouter une formation", color = Gold)
            }
        }

        Spacer(Modifier.height(80.dp))
    }
}

@Composable
fun EducationForm(onSave: (Education) -> Unit, onCancel: () -> Unit) {
    var school by remember { mutableStateOf("") }
    var degree by remember { mutableStateOf("") }
    var field by remember { mutableStateOf("") }
    var startYear by remember { mutableStateOf("") }
    var endYear by remember { mutableStateOf("") }

    Column(
        modifier = Modifier
            .fillMaxWidth()
            .background(MediumGray, RoundedCornerShape(12.dp))
            .padding(16.dp),
        verticalArrangement = Arrangement.spacedBy(12.dp)
    ) {
        Text("Nouvelle formation", fontWeight = FontWeight.SemiBold, color = Gold)

        CvTextField(value = degree, label = "Diplôme *") { degree = it }
        CvTextField(value = school, label = "École / Université *") { school = it }
        CvTextField(value = field, label = "Domaine / Spécialité") { field = it }
        Row(horizontalArrangement = Arrangement.spacedBy(12.dp)) {
            CvTextField(value = startYear, label = "Année début", onChange = { startYear = it }, modifier = Modifier.weight(1f))
            CvTextField(value = endYear, label = "Année fin", onChange = { endYear = it }, modifier = Modifier.weight(1f))
        }

        Row(horizontalArrangement = Arrangement.spacedBy(12.dp)) {
            OutlinedButton(onClick = onCancel, modifier = Modifier.weight(1f)) { Text("Annuler") }
            Button(
                onClick = {
                    if (degree.isNotBlank() && school.isNotBlank())
                        onSave(Education(school = school, degree = degree, field = field, startYear = startYear, endYear = endYear))
                },
                modifier = Modifier.weight(1f),
                colors = ButtonDefaults.buttonColors(containerColor = Gold)
            ) { Text("Enregistrer", color = MaterialTheme.colorScheme.onPrimary) }
        }
    }
}

@Composable
fun CvTextField(value: String, label: String, onChange: (String) -> Unit, modifier: Modifier = Modifier) {
    OutlinedTextField(
        value = value,
        onValueChange = onChange,
        label = { Text(label) },
        modifier = modifier.fillMaxWidth(),
        colors = OutlinedTextFieldDefaults.colors(focusedBorderColor = Gold, focusedLabelColor = Gold)
    )
}
